Simplified/SignInLogic/NYPLSignInBusinessLogic.swift

```diff
@@ -635,4 +635,18 @@ extension NYPLSignInBusinessLogic {
     }
   }
 
+  @objc func checkCardCreationEligibility(completion: @escaping () -> Void) {
+    guard let parentBarcode = self.resolveUserBarcode() else {
+      completion()
+      return
+    }
+    
+    let coordinator = juvenileCardCreationCoordinator ?? makeJuvenileCardCreationCoordinator(using: parentBarcode)
+    juvenileCardCreationCoordinator = coordinator
+
+    coordinator.checkJuvenileCreationEligibility(parentBarcode: parentBarcode) { [weak self] error in
+      self?.libraryAccount?.details?.setCardCreationEligibilityError(error: error)
```

Comment:
I don't know if it's correct to do this. Here you're modifying a library-related state (supportsCardCreator) based on the user's state (can the user create dependents?) but as far as I understand `supportsCardCreator` just means whether the library supports user creation via native card creator. I admit this was not documented very well, but it's what I can understand from the business logic class.

it might be cleaner to set the eligibility state as a new property on the business logic and then modify `juvenileCardsManagementIsPossible` to use it in addition to the other conditions.

Reply:
Right, thought the `supportsCardCreator` is an user's state for some reason, fixed